### PR TITLE
Modify matching algorithm

### PR DESCRIPTION
**Pull request by rafa238**
We added the command to manually execute the interview match and we modified the matching algorithm in the interview match

### CODE REVIEW

File: otter_buddy/cogs/interview_match.py [hunk 1]
```diff
@@ -121,15 +123,34 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
-    def make_pairs(self, week_otter_pool: list, placeholder: discord.Member):
+    def make_pairs(self, week_otter_pool: list, placeholder: discord.Member) -> List[Tuple[discord.Member, discord.Member]]:
         week_otter_pairs: list = []
+        week_otter_collaborator: list = []
+        week_otter_member: list = []
         if len(week_otter_pool) % 2 == 1:
             week_otter_pool.append(placeholder)
+
+        for otter in week_otter_pool:
+            str_otter_roles = [str(otter_role) for otter_role in otter.roles]
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:134]: Putting the type of the variable as prefix it's a bad practice, if you are using types already why not declaring this variable as a array of strings?

File: otter_buddy/cogs/interview_match.py [hunk 2]
```diff
@@ -121,15 +123,34 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
+                week_otter_collaborator.append(otter)
+            else:
+                week_otter_member.append(otter)
+
+        random.shuffle(week_otter_collaborator)
+        random.shuffle(week_otter_member)
+
+        for collaborator, member in list(zip(week_otter_collaborator, week_otter_member)):
+            week_otter_pairs.append((collaborator, member))
+            week_otter_collaborator.remove(collaborator)
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:145]: Why is it necessary to remove the collaborator and the members?

File: otter_buddy/cogs/interview_match.py [hunk 3]
```diff
@@ -121,15 +123,34 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
+            week_otter_member.remove(member)
         
-        random.shuffle(week_otter_pool)
-        for idx in range(len(week_otter_pool) // 2):
-            week_otter_pairs.append((week_otter_pool[idx*2], week_otter_pool[idx*2+1]))
+        for idx in range(len(week_otter_collaborator) // 2):
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:148]: Magic numbers are bad, what does 2 means?

File: otter_buddy/cogs/interview_match.py [hunk 4]
```diff
@@ -121,15 +123,34 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
+            week_otter_pairs.append((week_otter_collaborator[idx*2], week_otter_collaborator[idx*2+1]))
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:149]: Magic numbers

File: otter_buddy/cogs/interview_match.py [hunk 5]
```diff
@@ -65,8 +67,8 @@ async def send_weekly_message(self):
                 entry["message_id"] = message.id
                 db_interview_match.DbInterviewMatch.set_interview_match(entry)
     
-    async def check_weekly_message(self):
-        weekday = (datetime.datetime.today().weekday() - 1 + 7) % 7
+    async def check_weekly_message(self, weekday = None) -> None:
+        weekday = (datetime.datetime.today().weekday() - 1 + 7) % 7 if weekday is None else weekday
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:71]: Can we add a comment of the calculation that you are doing it here?

File: otter_buddy/cogs/interview_match.py [hunk 6]
```diff
@@ -178,6 +199,36 @@ async def stop(self, ctx):
...
+    invoke_without_command=True,
+    )   
+    @commands.has_any_role(OTTER_ADMIN, OTTER_MODERATOR)
+    async def run(self, ctx) -> None:
+        """
+        Admin commands related to trigger and test the activity
+        """
+        await ctx.send_help(ctx.command)
+
+    @run.command(brief="Admin command to trigger the send weekly message job")  # type: ignore
```
Comment (GoberInfinity) [inline otter_buddy/cogs/interview_match.py:214]: What the difference between this command and the one below, I can see the difference but looks like the documentation it's the same.

**Review by GoberInfinity — DISMISSED** [review]

File: otter_buddy/cogs/interview_match.py (hunk 1 above)
Comment (GoberInfinity) [reply otter_buddy/cogs/interview_match.py]: Any updates on this?

File: otter_buddy/cogs/interview_match.py [hunk 7]
```diff
@@ -65,8 +67,17 @@ async def send_weekly_message(self):
                 entry["message_id"] = message.id
                 db_interview_match.DbInterviewMatch.set_interview_match(entry)
     
-    async def check_weekly_message(self):
-        weekday = (datetime.datetime.today().weekday() - 1 + 7) % 7
+    async def check_weekly_message(self, weekday = None) -> None:
+        weekday = (datetime.datetime.today().weekday() - 1 + 7) % 7 if weekday is None else weekday
+        """
+        Above operation:
```
Comment (akotadi) [inline otter_buddy/cogs/interview_match.py:73]: Can you put the comment before the action (you can remove the "above operation"? Is better for reading it.

File: otter_buddy/cogs/interview_match.py [hunk 8]
```diff
@@ -121,14 +132,38 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
+        week_otter_member: list = []
         if len(week_otter_pool) % 2 == 1:
             week_otter_pool.append(placeholder)
-        
-        random.shuffle(week_otter_pool)
-        for idx in range(len(week_otter_pool) // 2):
-            week_otter_pairs.append((week_otter_pool[idx*2], week_otter_pool[idx*2+1]))
+
+        for otter in week_otter_pool:
+            otter_roles = [str(otter_role) for otter_role in otter.roles]
```
Comment (akotadi) [inline otter_buddy/cogs/interview_match.py:143]: Address the comment changing to this:
```suggestion
            otter_roles: list[str] = [str(otter_role) for otter_role in otter.roles]
```

File: otter_buddy/cogs/interview_match.py [hunk 9]
```diff
@@ -121,14 +132,38 @@ async def send_pair_message(self, otter_one: discord.User, otter_two: discord.Us
...
+        # Pair interviewees with colaborators
+        collaborator_index = 0
+        member_index = 0
+        while collaborator_index < len(week_otter_collaborator) and member_index < len(week_otter_member):
+            week_otter_pairs.append((week_otter_collaborator[collaborator_index], week_otter_member[member_index]))
+            collaborator_index += 1
+            member_index += 1
+
+        # Pair aditional colaborators
+        for idx in range(collaborator_index, len(week_otter_collaborator), 2):
```
Comment (akotadi) [inline otter_buddy/cogs/interview_match.py:161]: Nice change, I didn't know about the 3rd parameter

File: otter_buddy/cogs/interview_match.py (hunk 7 above)
Comment (rafa238) [reply otter_buddy/cogs/interview_match.py:73]: done

File: otter_buddy/cogs/interview_match.py (hunk 9 above)
Comment (rafa238) [reply otter_buddy/cogs/interview_match.py:161]: Thanks you :)

**Review by akotadi — APPROVED** [review]
LGTM! Thanks for doing this change :)

**Pull request merged**